detox/ios/Detox/Utilities/ViewHierarchyDump.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  UIWindow+ViewHierarchy.swift
+//  Detox
+//
+//  Created by Georgy Steshin on 18/08/2024.
+//  Copyright © 2024 Wix. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class ViewHierarchyDump {
+
+	static func getViewHierarchyDump(scene: UIWindowScene) -> String {
+		let windows = UIWindow.allKeyWindowSceneWindows
+
+		// Create the root ViewHierarchy element
+		var xml = "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
+		xml.append("\n<ViewHierarchy>")
+
+		// For each window we will dump the hierarchy
+		for window in windows {
+			//Ignore the touch visualizer window
+			if NSStringFromClass(type(of: window)) == "DTXTouchVisualizerWindow" {
+				continue
+			}
+
+			xml.append(dumpViewHierarchyRecursive(view: window, level: 1))
+		}
+
+		xml.append("\n</ViewHierarchy>")
+
+		return xml
```

Comment:
Small suggestion for refactoring, instead of building up a string incrementally and append values to it, I think it would be more semantic to build the different parts and then merge them

Reply:
that's the idea, same for the recursive one and android
added some functional programming
```swift
static func getViewHierarchyDump() -> String {
    let windows = UIWindow.allKeyWindowSceneWindows
    
    let xmlHeader = "<?xml version=\"1.0\" encoding=\"utf-8\"?>"
    let viewHierarchy = windows
        .filter { !($0 is DTXTouchVisualizerWindow) }
        .map { dumpViewHierarchyRecursive(view: $0, level: 1) }
        .joined()
    
    return """
    \(xmlHeader)
    <ViewHierarchy>
    \(viewHierarchy)
    </ViewHierarchy>
    """
}
```